spi/control-plane/contract-spi/src/main/java/org/eclipse/edc/connector/contract/spi/types/agreement/ContractAgreement.java

```diff
@@ -68,29 +68,25 @@ public String getId() {
     }
 
     /**
-     * The id of the data providing agent.
+     * The id of the data providing participant.
      * Please note that id should be taken from the corresponding data ecosystem.
-     * For example: In IDS the connector uses a URI from the IDS Information Model as ID. If the contract was
-     * negotiated inside the IDS ecosystem, this URI should be used here.
      *
      * @return provider id
      */
     @NotNull
-    public String getProviderAgentId() {
```

Comment:
wouldn't this be a breaking change? Maybe it would be better to rename the fields, adding new getters and builder methods and deprecating the old ones, it should be enough to avoid problems with the already persisted agreements (in the cosmos store, because in the sql one there will be no issue)

Reply:
Unfortunately, there's no way to avoid this due to the validity checks. Since we are not supporting backward compatibility, people will need to migrate their databases

Reply:
The SQL should not be a problem, for Cosmos, that we serialize/deserialize the `ContractAgreement` might.  
We could leave the old builder method in place depracated or if we choose to breake the changes here we need to remember this and put it in the release notes probably.

Reply:
Yeah, I think we should not keep the old builder methods because there is no way to get this to work without migrating the serialized JSON form anyway and having the methods can lead to people using them by accident in the future. This is basically a clean break along with the protocol changes.

Reply:
We definitely need to put this in the documentation along with how to configure the participant id. I've opened an issue for that here: #2887.